packages/flutter/lib/src/cupertino/route.dart

```diff
@@ -788,13 +788,38 @@ class _CupertinoBackGestureController<T> {
     }
 
     if (controller.isAnimating) {
+      late VoidCallback animationValueCallback;
+      late AnimationStatusListener animationStatusCallback;
+
+      // Keep the userGestureInProgress in true state so we don't change the
+      // curve of the page transition mid-flight since CupertinoPageTransition
+      // depends on userGestureInProgress. Switch the userGestureInProgress to
+      // false near the end, but just before so the app can be interacted with
+      // when the animation visibly ends.
+      animationValueCallback = () {
+        if (controller.value < 0.02) {
+          if (navigator.userGestureInProgress) {
```

Comment:
Do we need this check? Since this callback is called only once, this doesn't seem necessary?

Reply:
Probably not. This is a fragment from experimenting with other implementations that were unnecessarily complicated. I left it in for safety, but I don't think it would be necessary.